Userland/Libraries/LibJS/Runtime/Date.cpp

```diff
@@ -339,9 +339,11 @@ Value make_day(GlobalObject& global_object, Value year, Value month, Value date)
     // NOTE: This calculation has no side-effects and is unused, so we omit it
```

Comment:
```suggestion
    auto month = fmod(m, 12.0);
    auto mn = month >= 0.0 ? month : month + 12.0;
```

The month is supposed to be modulo-12.

Reply:
As the comment says this is not needed. The spec generates those values yes, but it doesn't tell you how to use them, and in our case we don't (since core datetime handles the > 12 month internally just fine.)

Reply:
It actually does not. See the relevant code from `tm_to_time`:
```cpp
    tm->tm_year += tm->tm_mon / 12;
    tm->tm_mon %= 12;
    if (tm->tm_mon < 0) {
        tm->tm_year--;
        tm->tm_mon += 12;
    }
```
If we pass it a negative month value, the year is decreased in our current implementation. While the JS spec says:
> The notation “x modulo y” (y must be finite and non-zero) computes a value k of the same sign as y (or zero) such that abs(k) < abs(y) and x - k = q × y for some integer q.

Therefore if the `month` parameter is negative, after the modulo operation, `mn` is required to have a positive value, therefore not decreasing the year.

Reply:
Doesn't matter, the temporal objects don't support negative months (with the exception of `Duration`, which is irrelevant here). Consider it an optimization :)